File: app/src/main/java/com/savvasdalkitsis/gameframe/ip/view/IpSetupView.java
package com.savvasdalkitsis.gameframe.ip.view;

import com.savvasdalkitsis.gameframe.ip.model.IpAddress;

public interface IpSetupView {

    void displayIpAddress(IpAddress ipAddress);

    void errorDiscoveringIpAddress(Throwable throwable);

    void addressSaved(IpAddress ipAddress);

    void displayDiscovering();

    void ipAddressDiscovered(IpAddress ipAddress);

    void tryingAddress(IpAddress ipAddress);

    void displayIdleView();
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/ip/view/IpSetupView.kt
package com.savvasdalkitsis.gameframe.ip.view

import com.savvasdalkitsis.gameframe.ip.model.IpAddress

interface IpSetupView {

    fun displayIpAddress(ipAddress: IpAddress)

    fun errorDiscoveringIpAddress(throwable: Throwable)

    fun addressSaved(ipAddress: IpAddress)

    fun displayDiscovering()

    fun ipAddressDiscovered(ipAddress: IpAddress)

    fun tryingAddress(ipAddress: IpAddress)

    fun displayIdleView()
}
